src/common/debug.rs

```diff
@@ -62,16 +62,15 @@ impl DebugState {
         {
             match patch {
                 DebugConfigPatch::Pyroscope(new_config) => {
-                    if let Some(mut pyroscope_state) = self.pyroscope.lock().take() {
-                        let stopped = pyroscope_state.stop_agent();
-                        if !stopped {
-                            return false;
-                        }
+                    let mut pyroscope_guard = self.pyroscope.lock();
+                    let pyroscope_state = pyroscope_guard.as_mut().unwrap();
```

Comment:
We should not unwrap here.

If the agent failed to start on construction, there will be a `None` in there causing a panic here.

Reply:
I'll quickly patch this one, as I see another point of improvement too.